### PR TITLE
Bug 1700891: OCP Branding

### PR DESCRIPTION
**Pull request by mrogers950**
https://bugzilla.redhat.com/show_bug.cgi?id=1700891

### CODE REVIEW

File: pkg/operator2/consoleconfig.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package operator2
```
Comment (enj) [inline pkg/operator2/consoleconfig.go]: drop every field that we do not actually use

File: pkg/operator2/operator.go [hunk 2]
```diff
@@ -105,6 +105,12 @@ const (
 	servingCertPathCert = servingCertMount + "/" + corev1.TLSCertKey
 	servingCertPathKey  = servingCertMount + "/" + corev1.TLSPrivateKeyKey
 
+	ocpBrandingSecretName  = "ocp-branding-template"
+	ocpBrandingSecretMount = systemConfigPathSecrets + "/" + ocpBrandingSecretName
+	ocpBrandingLoginKey    = "login.html"
```
Comment (enj) [inline pkg/operator2/operator.go]: use `configv1.LoginTemplateKey`, etc

File: pkg/operator2/deployment.go [hunk 3]
```diff
@@ -62,6 +62,12 @@ func defaultDeployment(
 			path:      routerCertsLocalMount,
 			keys:      sets.StringKeySet(routerSecret.Data).List(),
 		},
+		{
+			name:      ocpBrandingSecretName,
+			configmap: false,
+			path:      ocpBrandingSecretMount,
+			keys:      []string{ocpBrandingLoginKey, ocpBrandingProviderKey, ocpBrandingErrorKey},
```
Comment (enj) [inline pkg/operator2/deployment.go]: use `configv1.LoginTemplateKey`, etc

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,7 @@
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-authentication-operator
 COPY . .
-RUN go build -o authentication-operator ./cmd/authentication-operator
+RUN ADDITIONAL_GOTAGS="ocp" go build -o authentication-operator ./cmd/authentication-operator
```
Comment (enj) [inline Dockerfile]: This needs to be a go tag, and not an env var.

File: Dockerfile.rhel7 [hunk 5]
```diff
@@ -1,7 +1,7 @@
 FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.10 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-authentication-operator
 COPY . .
-RUN go build -o authentication-operator ./cmd/authentication-operator
+RUN ADDITIONAL_GOTAGS="ocp" go build -o authentication-operator ./cmd/authentication-operator
```
Comment (enj) [inline Dockerfile.rhel7]: same

File: pkg/operator2/oauth.go [hunk 6]
```diff
@@ -2,6 +2,10 @@ package operator2
 
 import (
 	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
```
Comment (enj) [inline pkg/operator2/oauth.go]: now sure if this is the right lib to use

File: pkg/operator2/oauth.go [hunk 7]
```diff
@@ -191,3 +205,33 @@ func defaultOAuthConfig(oauthConfig *configv1.OAuth) *configv1.OAuth {
 
 	return out
 }
+
+func (c *authOperator) handleOCPBrandingSecret() (*corev1.Secret, error) {
```
Comment (enj) [inline pkg/operator2/oauth.go]: move branding to its own file

File: pkg/operator2/oauth.go [hunk 8]
```diff
@@ -59,11 +63,21 @@ func (c *authOperator) handleOAuthConfig(
...
 		templates = &osinv1.OAuthTemplates{
 			Login:             syncData.addTemplateSecret(configTemplates.Login, loginField, configv1.LoginTemplateKey),
 			ProviderSelection: syncData.addTemplateSecret(configTemplates.ProviderSelection, providerSelectionField, configv1.ProviderSelectionTemplateKey),
 			Error:             syncData.addTemplateSecret(configTemplates.Error, errorField, configv1.ErrorsTemplateKey),
 		}
+	} else {
+		if c.useOCPBranding() || strings.Contains(os.Getenv("ADDITIONAL_GOTAGS"), "ocp") {
+			klog.Infof("DBG: using ocp templates")
+			templates = &osinv1.OAuthTemplates{
+				Login:             ocpBrandingSecretMount + "/" + ocpBrandingLoginKey,
```
Comment (enj) [inline pkg/operator2/oauth.go]: make these 3 consts

File: pkg/operator2/oauth.go [hunk 9]
```diff
@@ -59,11 +63,21 @@ func (c *authOperator) handleOAuthConfig(
...
 	emptyTemplates := configv1.OAuthTemplates{}
 	if configTemplates := oauthConfig.Spec.Templates; configTemplates != emptyTemplates {
+		klog.Infof("DBG: using spec templates")
 		templates = &osinv1.OAuthTemplates{
 			Login:             syncData.addTemplateSecret(configTemplates.Login, loginField, configv1.LoginTemplateKey),
 			ProviderSelection: syncData.addTemplateSecret(configTemplates.ProviderSelection, providerSelectionField, configv1.ProviderSelectionTemplateKey),
 			Error:             syncData.addTemplateSecret(configTemplates.Error, errorField, configv1.ErrorsTemplateKey),
 		}
+	} else {
+		if c.useOCPBranding() || strings.Contains(os.Getenv("ADDITIONAL_GOTAGS"), "ocp") {
```
Comment (enj) [inline pkg/operator2/oauth.go]: no env var, use build tag

File: pkg/operator2/oauth.go [hunk 10]
```diff
@@ -191,3 +205,33 @@ func defaultOAuthConfig(oauthConfig *configv1.OAuth) *configv1.OAuth {
 
 	return out
 }
+
+func (c *authOperator) handleOCPBrandingSecret() (*corev1.Secret, error) {
+	return c.secrets.Secrets(targetNamespace).Get(ocpBrandingSecretName, metav1.GetOptions{})
+}
+
+func (c *authOperator) useOCPBranding() bool {
```
Comment (enj) [inline pkg/operator2/oauth.go]: this is not fully correct - need to handle OKD correctly

File: pkg/operator2/oauth.go [hunk 11]
```diff
@@ -59,11 +63,21 @@ func (c *authOperator) handleOAuthConfig(
 
 	emptyTemplates := configv1.OAuthTemplates{}
 	if configTemplates := oauthConfig.Spec.Templates; configTemplates != emptyTemplates {
+		klog.Infof("DBG: using spec templates")
 		templates = &osinv1.OAuthTemplates{
 			Login:             syncData.addTemplateSecret(configTemplates.Login, loginField, configv1.LoginTemplateKey),
 			ProviderSelection: syncData.addTemplateSecret(configTemplates.ProviderSelection, providerSelectionField, configv1.ProviderSelectionTemplateKey),
 			Error:             syncData.addTemplateSecret(configTemplates.Error, errorField, configv1.ErrorsTemplateKey),
 		}
+	} else {
```
Comment (enj) [inline pkg/operator2/oauth.go]: move all of this logic out to its own file

and you need to handle each user configuration separately - you can set just one of the templates

**Comment by mrogers950** [comment]
@enj updated to address your comments

**Pull request merged**